src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/FindPaginationTableIntegrationTest.java

```diff
@@ -30,12 +39,134 @@ public final void dropScenario() {
   }
 
   @Test
-  public void findWithPageState() {
+  @Order(1)
+  public void findWithNoSortAndPartitionedPageState() throws Exception {
+    // This test goes through all the pages in each partition and verifies the content
+    // It also checks no sort clause
+    for (int partition = 0; partition < 3; partition++) {
+      String partitionedKeyValue = "partition-" + partition;
+      FILTER_ID =
+          ImmutableMap.of(fieldName(ThreeClusteringKeysTableScenario.ID_COL), partitionedKeyValue);
+      String pageState =
+          assertTableCommand(keyspaceName, TABLE_NAME)
+              .templated()
+              .find(FILTER_ID, List.of(), null) // find all rows in each partition
+              .wasSuccessful()
+              .body("data.documents", hasSize(20)) // Verify that 20 rows are returned.
+              .body(
+                  "data.documents",
+                  jsonEquals(
+                      SCENARIO.getPageContent(0, 20, partitionedKeyValue))) // verify the content
+              .hasNextPageState()
+              .extractNextPageState();
+
+      Map<String, Object> options = ImmutableMap.of("pageState", pageState);
+
+      // The next page should have 10 rows and no page state
+      assertTableCommand(keyspaceName, TABLE_NAME)
+          .templated()
+          .find(FILTER_ID, List.of(), null, options)
+          .wasSuccessful()
+          .body("data.documents", hasSize(10)) // 10 rows for the last page
+          .body(
+              "data.documents",
+              jsonEquals(
+                  SCENARIO.getPageContent(20, 20, partitionedKeyValue))) // verify the content
+          .doesNotHaveNextPageState();
+    }
+  }
+
+  @Test
+  @Order(2)
+  public void findWithEmptySortAndPageState() {
+    // This test checks the empty sort (equals no sort clause)
```

Comment:
good to have comments like this, better to use javadoc standard for a function with /**